packages/connect/src/device/Device.ts

```diff
@@ -336,37 +336,8 @@ export class Device extends TypedEmitter<DeviceEvents> {
 
             // update features
             try {
-                if (fn) {
-                    await this.initialize(
-                        !!options.useEmptyPassphrase,
-                        !!options.useCardanoDerivation,
-                    );
-                } else {
-                    // do not initialize while firstRunPromise otherwise `features.session_id` could be affected
-                    await Promise.race([
```

Comment:
i think that the
```
 if(fn) else
```
part needs to stay there. but the promise.race and this.incosistent can probably go away

Reply:
If we remove the `Promise.race` part it does not make sense to have the `else`. What about:

```
if (fn) {
   await this.initialize(!!options.useCardanoDerivation);
}
```

:question:

Reply:
```
if (fn) { 
    await this.initialize(!!options.useCardanoDerivation);
} else {
   await this.getFeatures(),
}
```

should it be